### PR TITLE
Create a database key to fix name collusion on loading captures

### PR DESCRIPTION
**Pull request by yalcinmelihyasin**
http://b/139294646

### CODE REVIEW

File: gapis/capture/capture.proto [hunk 1]
```diff
@@ -33,9 +33,11 @@ message File {
 // Record holds all the data for an entire capture.
 message Record {
   // Name of the capture.
-  string name = 1;
+  string key = 1;
```
Comment (yalcinmelihyasin) [inline gapis/capture/capture.proto]: This is introduced because name is being used for other reasons then being a database key as well.

Comment (hevrard) [reply gapis/capture/capture.proto]: The comment above is not matching the code anymore, can you edit the comment to:
`Key to index the capture`

File: gapis/server/server.go [hunk 2]
```diff
@@ -233,10 +233,18 @@ func (s *server) LoadCapture(ctx context.Context, path string) (*path.Capture, e
...
 	}
-	name := filepath.Base(path)
 
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return nil, err
+	}
+
+	name := fileInfo.Name()
+	key := fmt.Sprintf("%v%v%v", name, fileInfo.Size(), fileInfo.ModTime().Unix())
```
Comment (hevrard) [inline gapis/server/server.go]: Add a comment saying something like:
```
// The key contains more information than the file name in order to disambiguate capture that have a similar file name.
```

File: gapis/server/server.go [hunk 3]
```diff
@@ -233,10 +233,19 @@ func (s *server) LoadCapture(ctx context.Context, path string) (*path.Capture, e
...
 
+	fileInfo, err := os.Stat(path)
+	if err != nil {
+		return nil, err
+	}
+
+	name := fileInfo.Name()
+	// Create a key to prevent name collusion between traces.
+	key := fmt.Sprintf("%v%v%v", name, fileInfo.Size(), fileInfo.ModTime().Unix())
+	key = strings.ReplaceAll(key, " ", "")
```
Comment (pmuetschard) [inline gapis/server/server.go]: what's wrong with spaces?

File: gapis/capture/capture.proto [hunk 4]
```diff
@@ -32,10 +32,12 @@ message File {
...
 // Record holds all the data for an entire capture.
 message Record {
-  // Name of the capture.
-  string name = 1;
+  // Key for storing the capture.
+  string key = 1;
   // Database identifier of the data.
   bytes data = 2;
+  // Name of the capture.
+  string capture_name = 3;
```
Comment (pmuetschard) [inline gapis/capture/capture.proto]: Why rename the field?

File: gapis/capture/capture.go [hunk 5]
```diff
@@ -129,14 +129,16 @@ func ResolvePerfettoFromPath(ctx context.Context, p *path.Capture) (*PerfettoCap
 }
 
 // Import imports the capture by name and data, and stores it in the database.
-func Import(ctx context.Context, name string, src Source) (*path.Capture, error) {
+func Import(ctx context.Context, name string, key string, src Source) (*path.Capture, error) {
```
Comment (ben-clayton) [inline gapis/capture/capture.go]: Why don't we just use a hash of the capture's data?

Comment (yalcinmelihyasin) [reply gapis/capture/capture.go]: Hashing can take long time depending on the capture size which can be in GBs.

File: gapis/server/server.go (hunk 3 above)
Comment (yalcinmelihyasin) [reply gapis/server/server.go]: Done.

File: gapis/capture/capture.proto (hunk 4 above)
Comment (yalcinmelihyasin) [reply gapis/capture/capture.proto]: reverted back.

File: gapis/capture/capture.go (hunk 5 above)
Comment (yalcinmelihyasin) [reply gapis/capture/capture.go]: Merging this version and if we decide to use hash we can update it in another PR.

**Pull request merged**